src/app-layer.c

```diff
@@ -80,6 +91,14 @@ static inline int ProtoDetectDone(const Flow *f, const TcpSession *ssn, uint8_t
             (FLOW_IS_PM_DONE(f, direction) && FLOW_IS_PP_DONE(f, direction)));
 }
 
+static void AppLayerIncFlowCounter(uint8_t ipproto, AppProto alproto, ThreadVars *tv)
+{
+    if (tv) {
```

Comment:
I think tv will only be NULL in unittests, so it may be worth only doing this check then. Otherwise, add 'likely'